core/src/main/java/com/graphhopper/routing/Path.java

```diff
@@ -703,7 +704,16 @@ private boolean surroundingStreetsAreSlowerByFactor(int baseNode, int prevNode,
                 double maxSurroundingSpeed = -1;
                 double tmpSpeed;
                 while (edgeIter.next()) {
-                    tmpSpeed = encoder.getSpeed(edgeIter.getFlags());
+                    if (encoder instanceof DataFlagEncoder) {
+                        tmpSpeed = ((DataFlagEncoder) encoder).getMaxspeed(edgeIter, 0, false);
```

Comment:
Might be, yes. I am a bit afraid of rounding issues, as I am comparing these values, but sure, I can try that. OTOH I think we need a different logic for the DataFlagEncoder. We need to consider the highway tag and speed becomes rather unimportant (maybe we only need to check speed with a low priority, whereas for the other encoders speed is important for the instruction generation.